src/server/frontend_wayland/wlr_screencopy_v1.cpp

```diff
@@ -20,18 +20,38 @@
 #include "mir/graphics/graphic_buffer_allocator.h"
 #include "mir/renderer/sw/pixel_source.h"
 #include "mir/graphics/buffer.h"
+#include "mir/scene/scene_change_notification.h"
+#include "mir/frontend/surface_stack.h"
+#include "mir/geometry/rectangles.h"
 #include "mir/executor.h"
 #include "wayland_wrapper.h"
 #include "wayland_timespec.h"
 #include "output_manager.h"
 
 #include <boost/throw_exception.hpp>
+#include <mutex>
+#include <optional>
 
 namespace mf = mir::frontend;
 namespace mw = mir::wayland;
 namespace mg = mir::graphics;
+namespace ms = mir::scene;
 namespace geom = mir::geometry;
 
+namespace
+{
+struct FrameParams
+{
+    geom::Rectangle area;
+    wl_resource* output;
+
+    auto operator==(FrameParams const& other) const -> bool
+    {
+        return area == other.area && output == other.output;
+    }
```

Comment:
It is a shame we support compilers without
```suggestion
    bool operator==(FrameParams const&) = default;
```
... also and the "spaceship operator"